oauth2_client/client.py

```diff
@@ -43,9 +94,30 @@ def request(self, method, url, *args, **kwargs):  # pylint: disable=arguments-di
 
         Returns:
             Response object.
+
+        Raises:
+            TokenExpiredError: when token expiry detected and `expired_callback` not specified. Also
+                when 403 response code received.
+
+            oauth2_client.exceptions.TokenError: when `expired_callback` was specified and more
+                than one token re-fetch happens within `REAUTH_LIMIT_SECONDS` period
         """
         absolute_url = urljoin(self.service_host, url)
-        return super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
+        try:
+            # super().request can raise `TokenExpiredError`
+            resp = super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
+            if resp.status_code == 403:
+                raise TokenExpiredError(description="403 status code response received")
+            else:
+                return resp
+        except TokenExpiredError:
+            log.debug("Access Token for app=%s expired.", self.app.name)
+            if not self.expired_callback:
+                raise
```

Comment:
after discussing with @labstersteve I removed the custom exception from the code entirely and sticked to the 3rd party's `TokenExpiredError`. Less code to maintain and same information passed in case something goes wrong.